src/appleseed/renderer/utility/paramarray.h

```diff
@@ -578,9 +578,9 @@ bool ParamArray::contains(
     const StringVec&            allowed_values,
     const T&                    value)
 {
-    for (foundation::const_each<StringVec> i = allowed_values; i; ++i)
+    for (auto const &i : allowed_values)
```

Comment:
`const auto& i` (ampersand goes on the type's side).

But as I said above, `auto` should only be used when the type isn't known, or is particularly cumbersome.